src/components/EditTeamModal/index.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react';
+import { PropTypes as PT } from 'prop-types';
+import Container from './container';
+import { Modal } from '../../components/common';
+import { FormStyleContainer } from '../common_styled/FormStyleContainer';
+import { withFormik } from 'formik';
+
+const FormikEnhancer = withFormik({
+  displayName: 'Edit Team Form',
+
+  // This updates the form when props change.
+  enableReinitialize: true,
+
+  // validateOn
+
+  mapPropsToValues: ({ team }) => ({
+    teamName: team ? team.teamName : '',
+  }),
+
+  // Custom sync validation
+  validate: values => {
+    const { teamName } = values;
+    let errors = {};
+    if (teamName === '') {
+      errors.teamName = 'Team name cannot be empty.';
+    }
+    return errors;
+  },
+
+  handleSubmit: (payload, bag) => {
+    bag.props.handleFormSubmit(bag.props.team, payload);
+  },
+});
+
+class EditTeamModal extends Component {
+  static propTypes = {
+    team: PT.object,
```

Comment:
Recommend using PT.shape instead of PT.object to better understand the team/values/errors objects